test/integration/create_user_test.rb

```diff
@@ -7,43 +7,74 @@
 
 require 'test_helper'
 class CreateUserTest < ActionDispatch::IntegrationTest
-include IntegrationTest::DefaultHelper
+  include IntegrationTest::DefaultHelper
 
   test 'admin creates new user' do
     login_as('admin')
     post admin_users_path, params: { user: {
-                                  username: "simon",
-                                  password: "password",
-                                  admin: 0,
-                                  givenname: "Simon",
-                                  surname: "Kern"} }
+      username: 'simon',
+      password: 'password',
+      admin: 0,
+      givenname: 'Simon',
+      surname: 'Kern'
+    } }
     assert_redirected_to admin_users_path
     assert User.find_by_username('simon')
     logout
     login_as('simon')
   end
+  
+  test 'admin creates new user with username with . - _' do
+    login_as('admin')
+    post admin_users_path, params: { user: {
+      username: 'simon.s-s_s',
+      password: 'password',
+      admin: 0,
+      givenname: 'Simon',
+      surname: 'Kern'
+    } }
+    assert_redirected_to admin_users_path
+    assert User.find_by_username('simon.s-s_s')
+    logout
+    login_as('simon.s-s_s')
+  end
+
+  test 'admin cannot create user with special chars' do
+    login_as('admin')
+    post admin_users_path, params: { user: {
+      username: 'simon?',
+      password: 'password',
+      admin: 0,
+      givenname: 'Simon',
+      surname: 'Kern'
+    } }
+    assert_equal request.fullpath, admin_users_path
```

Comment:
do not redirect to admin users path ....
re-render user edit/new if error. use model validations !!!!